src/Guard.php

```diff
@@ -267,12 +272,29 @@ protected function enforceStorageLimit()
             return;
         }
 
-        if (!is_array($this->storage) && !$this->storage instanceof Countable) {
+        // $storage must be an array or implement Countable and either Iterator or IteratorAggregate
+        if (!is_array($this->storage)
+            && !($this->storage instanceof Countable
+                && ($this->storage instanceof Iterator || $this->storage instanceof IteratorAggregate))
```

Comment:
Could the check for `Iterator` and `IteratorAggregate` be replaced by `Traversable`?